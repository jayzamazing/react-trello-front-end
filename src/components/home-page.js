import React from 'react';
import {Link, Redirect} from 'react-router-dom';
import {connect} from 'react-redux';
import './home.css';

class HomePage extends React.Component {
  componentWillMount() {
    document.body.style.backgroundColor = "#026aa7";
  }
  componentWillUnmount() {
    document.body.style.backgroundColor = null;
  }
  render() {
    // If we are logged in redirect straight to the user's dashboard
    if (this.props.loggedIn) {
        return <Redirect to="/boards" />;
    }
    return (
      <section>
        <main className="home container">
          <div className="row wrapper">
            <div className="text-center">
              <h1>Welcome to Trello!</h1>
            </div>
            <div className="text-center wrapper-2">
              <h3>This is a simple project that emulates some of trello's functionality.</h3>
              <div className="wrapper-2">
                <Link to="/register">
                  <button className="btn sign-up-btn">
                    <h4>Sign Up - It's Free.</h4>
                    </button>
                </Link>
              </div>
              <div className="wrapper">
                <p>Already use Trello? <Link to="/login" className="a-login">Log in.</Link></p>
              </div>
            </div>
          </div>
        </main>
      </section>
    );
  }
}
const mapStateToProps = state => ({
    loggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(HomePage);
