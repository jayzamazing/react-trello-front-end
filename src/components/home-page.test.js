'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {HomePage} from './home-page';
import {Redirect} from 'react-router-dom';

describe('home-page', () => {
  it('should render page', () => {
    shallow(<HomePage />);
  });
  it('snapshot', () => {
    const wrapper = shallow(<HomePage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should redirect when user is logged in', () => {
    const wrapper = shallow(<HomePage loggedIn="true"/>);
    expect(wrapper.find(Redirect)).toHaveLength(1);
  });
  it('redirect snapshow', () => {
    const wrapper = shallow(<HomePage loggedIn="true"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
