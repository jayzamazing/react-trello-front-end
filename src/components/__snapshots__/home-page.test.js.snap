// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home-page redirect snapshow 1`] = `
<Redirect
  push={false}
  to="/boards"
/>
`;

exports[`home-page snapshot 1`] = `
<section>
  <main
    className="home container"
  >
    <div
      className="row wrapper"
    >
      <div
        className="text-center"
      >
        <h1>
          Welcome to Trello!
        </h1>
      </div>
      <div
        className="text-center wrapper-2"
      >
        <h3>
          This is a simple project that emulates some of trello's functionality.
        </h3>
        <div
          className="wrapper-2"
        >
          <Link
            replace={false}
            to="/register"
          >
            <button
              className="btn sign-up-btn"
            >
              <h4>
                Sign Up - It's Free.
              </h4>
            </button>
          </Link>
        </div>
        <div
          className="wrapper"
        >
          <p>
            Already use Trello? 
            <Link
              className="a-login"
              replace={false}
              to="/login"
            >
              Log in.
            </Link>
          </p>
        </div>
      </div>
    </div>
    <div
      className="row wrapper-2 video"
    >
      <iframe
        allow="encrypted-media"
        allowFullScreen={true}
        frameBorder="0"
        gesture="media"
        height="315"
        src="https://www.youtube.com/embed/GQuNxPiQWTY?rel=0"
        title="trello walk through"
        width="560"
      />
    </div>
  </main>
</section>
`;
